src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -209,14 +209,21 @@ function toggleCollapse(e, pans) {
     var panels = pans || $('div.panel-collapse');
 
     if ($(e).html().trim().startsWith(expand)) {
-        isExpandingAll = true;
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
+        $('.ajax-response-auto').click();
+        var $autoLoadCount = $('.ajax_auto').click();
+        var $isAutoLoadingTriggered = $autoLoadCount.length !== 0;
+
+        if (!$isAutoLoadingTriggered) {
+            isExpandingAll = true;
+            var i = 0;
+            for (var idx = 0; idx < panels.length; idx++) {
+                if ($(panels[idx]).attr('class').indexOf('in') === -1) {
+                    setTimeout(showSingleCollapse, 50 * i, panels[idx]);
```

Comment:
This a little more advanced, @li-kai @thenaesh can either of you a d.contributor issue regarding this? The original person who wrote this did not understand hoisting well enough, in the issue can we improve this by binding idx to setTimeout instead of maintaining a separate i variable?

http://ignaciothayer.com/post/a-dangerous-example-of-javascript-hoisting/